sr_speech_control/src/sr_speech_control/speech_control.py

```diff
@@ -15,66 +15,46 @@
 # with this program. If not, see <http://www.gnu.org/licenses/>.
 
 import rospy
-import time
 import speech_recognition as sr
-from difflib import get_close_matches
 from std_msgs.msg import String
 
 
 class SpeechControl(object):
-    def __init__(self, trigger_word, command_words, command_topic='speech_control',
-                 non_speaking_duration=0.1, pause_threshold=0.2):
+    def __init__(self, trigger_word, command_topic='sr_speech_control'):
+        self.microphone = sr.Microphone()
+        for i, microphone_name in enumerate(sr.Microphone.list_microphone_names()):
+            if "pulse" in microphone_name:
+                self.microphone = sr.Microphone(device_index=i)
+                rospy.loginfo("Using microphone: {}".format(microphone_name))
         self.trigger_word = trigger_word
-        self.command_words = command_words
         self.recognizer = sr.Recognizer()
-        self.microphone = sr.Microphone()
         self.command_publisher = rospy.Publisher(command_topic, String, queue_size=1)
-        self.command_to_be_executed = None
-
-        self._init_recognizer(non_speaking_duration, pause_threshold)
         self._stop_listening = self.recognizer.listen_in_background(self.microphone, self._recognizer_callback)
 
-    def _init_recognizer(self, non_speaking_duration, pause_threshold):
-        with self.microphone as source:
-            self.recognizer.adjust_for_ambient_noise(source)
```

Comment:
What is an advantage of removing this?

Reply:
I think it is not practical to assume that noise during node startup would be representative noise. If we really want to do this we should notify user: now be quiet as we are calibrating microphone for ambient noise, if there is any unusual noise please re-calibrate later manually or something similar.

Reply:
OK, makes sense. What about the parameters below?

Reply:
As I mentioned in PR description I couldn't find `non_speaking_duration` option in documentation (https://github.com/Uberi/speech_recognition/blob/master/reference/library-reference.rst) therefore removed it.
Also default value of `pause_threshold` worked better for me so I decided not to tweak anything and just go with defaults until we find hard evidence that customisation is necessary.

Reply:
It was improving performance for me, effectively reducing the lag in usage (when using the default the recognizer waits more time after user stops speaking befire starting the recognition). Reduction in lag/latency was one of the original and most voiced concerns. Regarding the non_speakin_duration:
https://github.com/Uberi/speech_recognition/blob/c89856088ad81d81d38be314e3db50905481c5fe/speech_recognition/__init__.py#L512

It needs to be parametrized as well if we want to reduce pause_threshold, due to this:
https://github.com/Uberi/speech_recognition/blob/c89856088ad81d81d38be314e3db50905481c5fe/speech_recognition/__init__.py#L557

Reply:
Re-added `non_speakin_duration` and `pause_threshold` as parameters. By default they are not set because setting them to your values doesn't work on my machine at all. Or maybe I just speak too slow :)